Replace 'import ocean.transition' inside ocean.core modules

don-clugston-sociomantic opened the pull request:
The `ocean.transition` module was used during D2 transition. Some of the aliases which it defined (`istring`, `cstring`, `mstring`) turned out to be useful, and were moved to another module.  This PR updates the imports to point to the new module. A couple of modules used `gc_stats` which doesn't have a replacement, but in those cases I made the transition import into a local import in a single function.

This PR cleans up all the modules in the `ocean.core.*` package. The rest of ocean will have to be updated at some time in the future.

don-clugston-sociomantic committed:
Reduce transition imports in modules using gc_usage

It is not possible to completely remove the ocean.transition import from
these modules, because they use gc_usage(), which exists only in
ocean.transition. But the import can be moved inside the functions which
actually use this function.

stefan-koch-sociomantic reviewed (approved):
Small inconsistencies in the imports but it does not matter either way.

stefan-koch-sociomantic commented on src/ocean/io/console/AppStatus.d:
```diff
@@ -907,6 +907,8 @@ public class AppStatus
 
     public bool getMemoryUsage ( out float mem_allocated, out float mem_free )
     {
+        import ocean.transition;
```
this could and should be replaced with core.memory.

don-clugston-sociomantic committed:
Make imports of 'Octal' explicit

These modules use ocean.transition.Octal, which does not exist
elsewhere in ocean, preventing removal of the ocean.transition import.

don-clugston-sociomantic commented:
I have split this into two different PRs.
